packages/playwright-utils/src/emits/index.ts

```diff
@@ -0,0 +1,52 @@
+import { expect } from "@playwright/test";
+import type { Component } from "vue";
+import type { ComponentEmitHandler } from "./types.js";
+
+export const EMIT_SPY_SYMBOL = Symbol("EMIT_SPY_SYMBOL");
+
+/**
+ * Creates a simple, typed spy for recording emits.
+ *
+ * @example
+ * ```tsx
+ * // create spy
+ * const onUpdateOpen = createEmitSpy<typeof OnyxColorSchemeDialog, "onUpdate:open">();
+ * // add spy
+ * const component = await mount(<OnyxColorSchemeDialog onUpdate:open={onUpdateOpen} />);
+ * // check spy
+ * expectEmit(onUpdateOpen, 1, [false]);
+ * ```
+ */
+export const createEmitSpy = <
+  C extends Component,
+  Key extends keyof Emits,
+  Emits = ComponentEmitHandler<C>,
+  Handler = NonNullable<Emits[Key]>,
+  Args extends unknown[] = Handler extends (...args: infer _Args) => unknown ? _Args : unknown[],
+>() => {
+  const calls: Args[] = [];
+  const handler = (...args: Args) => {
+    calls.push(args);
+  };
+  handler[EMIT_SPY_SYMBOL] = calls;
+  return handler;
+};
+
+/**
+ * Asserts the emits recorded from the spy created by `createEmitSpy`.
+ * Expects the spy to have recorded `n` calls, and `matches` to match the arguments of the last call.
+ * @see `createEmitSpy` documentation for example usage.
+ */
+export const expectEmit = <Handler extends { [EMIT_SPY_SYMBOL]: unknown[][] }>(
+  emitSpy: Handler,
+  n: number,
+  matches: unknown[] | Record<string, unknown>,
```

Comment:
```suggestion
  n: number,
  matches: unknown[] | Record<string, unknown>,
```

Can't we combine the second and third parameter? So that you just pass in an array of expected data and the length of the array is the expected n?

Reply:
I like this approach, because it allows readable step-by-step testing, e.g.:

```tsx
// ACT
  await clickOption("Light");
  await component.getByRole("button", { name: "Apply" }).click();

  // ASSERT
  expect(updateModelValueEvents).toStrictEqual(["dark", "auto", "light"]);
  expectEmit(onUpdateOpen, 3, [false]);

  // ACT
  await component.getByRole("button", { name: "Cancel" }).click();

  // ASSERT
  expectEmit(onUpdateOpen, 4, [false]);

  // ACT
  await page.keyboard.press("Escape");

  // ASSERT
  expectEmit(onUpdateOpen, 5, [false]);
```